huaweicloud/services/cts/resource_huaweicloud_cts_data_tracker.go

```diff
@@ -199,8 +210,16 @@ func buildDataBucketOpts(d *schema.ResourceData) *cts.DataBucket {
 
 func buildTransferBucketOpts(d *schema.ResourceData) *cts.TrackerObsInfo {
 	transferCfg := cts.TrackerObsInfo{
-		BucketName:     utils.String(d.Get("bucket_name").(string)),
-		FilePrefixName: utils.String(d.Get("file_prefix").(string)),
+		BucketName:      utils.String(d.Get("bucket_name").(string)),
+		FilePrefixName:  utils.String(d.Get("file_prefix").(string)),
+		IsSortByService: utils.Bool(d.Get("is_sort_by_service").(bool)),
+	}
+	if v, ok := d.GetOk("compress_type"); ok {
+		compressType := cts.GetTrackerObsInfoCompressTypeEnum().GZIP
+		if v.(string) != "gzip" {
+			compressType = cts.GetTrackerObsInfoCompressTypeEnum().JSON
+		}
```

Comment:
how about add a validation to the param of compress_type, if the value is not gzip, then the value will be json, no matter whether the value is json